pabcnetc/ConsoleCompiler.cs

```diff
@@ -433,77 +455,81 @@ public static int Main(string[] initialArgs)
                     StringResourcesLanguage.CurrentLanguageName = StringResourcesLanguage.AccessibleLanguages[StringResourcesLanguage.TwoLetterISOLanguages.IndexOf("en")];
                 else
                     StringResourcesLanguage.CurrentLanguageName = StringResourcesLanguage.AccessibleLanguages[StringResourcesLanguage.TwoLetterISOLanguages.IndexOf(StringResourcesLanguage.CurrentTwoLetterISO)];
-                
+
             }
-                
 
-            for (int i = 0; i < ConsoleBufferWidth-1; i++)
-              BlankString += " ";
+
+            BlankString = new string(' ', ConsoleBufferWidth - 1);
 
             Console.ForegroundColor = ConsoleColor.White;
             OutputDirectory="";
             
             Console.Title = StringResourcesGet("STARTING");
             Reset();
             Console.Title = Compiler.Banner;
-            
-
-            //Console.CursorLeft = (Console.BufferWidth - Compiler.Banner.Length) / 2;
-            //Console.WriteLine(Compiler.Banner); 
 
 
-            short_output = args.Count != 1;
+            //Console.CursorLeft = (Console.BufferWidth - Compiler.Banner.Length) / 2;
+            //Console.WriteLine(Compiler.Banner);
 
-            if (args.Count >= 1) 
+            if (args.Count > 0)
             {
-                FileName = args[0];
-            }
-            if (args.Count >= 2)
-                if (args[1] == "/rebuildnodebug")
+                List<FileInfo> fileInfo = new List<FileInfo>();
+                for (int i = 0; i < args.Count; i++)
                 {
-                    Rebuild = true;
-                    Compiler.InternalDebug.IncludeDebugInfoInPCU = false;
-                    short_output = false;
-                    DetailOut = true;
-                }
-                else
-                    if (args[1] == "/rebuild")
+                    if (args[i] == "/rebuildnodebug")
+                    {
+                        Rebuild = true;
+                        Compiler.InternalDebug.IncludeDebugInfoInPCU = false;
+                        short_output = false;
+                        DetailOut = true;
+                    }
+                    else if (args[i] == "/rebuild")
                     {
                         Rebuild = true;
                         short_output = false;
                         DetailOut = true;
                     }
                     else
-                        TagertFileName = args[1];
-            outputType = CompilerOptions.OutputType.ConsoleApplicaton;
-            if (args.Count >= 3)
-                    SetOutType(args[2]);
-            
-
-
-            ShowConnectedParsers();
-            ShowConnectedConversions();
-            if (args.Count >= 1)
-            {
-                return CompileAssembly(FileName, outputType, true);
+                    {
+                        if (File.Exists(args[i]) || File.Exists(Environment.CurrentDirectory + args[i]))
```

Comment:
I guess the second check would be for `\FileName.pas`, which `File.Exists` would check at `C:\FileName.pas` (current root drive instead of `C:\`) and not in the current directory.
But it would also awkwardly concat the names of the last directory and the file if `args[i]` does not start with `\`.
Either way, this looks like a hack, could be a lot clearer with `Path.Combine` or at least a comment.
And do we need to check that kind of file in the current directory, instead of the root, as the system does by default?

Reply:
I think this is important in the case of running the compiler from cmd, where WorkingDirectory is the folder with the files to compile, so if you call
C:\...\pabcnetc.exe test.pas
Then the file should be checked in the current directory.
I may have misunderstood you.